packages/ra-ui-materialui/src/button/DeleteButton.tsx

```diff
@@ -54,8 +56,19 @@ export const DeleteButton = <RecordType extends RaRecord = any>(
 ) => {
     const { mutationMode, ...rest } = props;
     const record = useRecordContext(props);
+    const resource = useResourceContext(props);
+    if (!resource) {
+        throw new Error(
+            '<DeleteButton> components should be used inside a <Resource> component or provided the resource prop.'
+        );
+    }
```

Comment:
Because I added this, I had to update many tests that used `<SimpleForm>` as it has a `<DeleteButton>` in its toolbar